azure/services/virtualmachines/virtualmachines.go

```diff
@@ -138,6 +153,36 @@ func (s *Service) Delete(ctx context.Context) error {
 	return err
 }
 
+func (s *Service) checkUserAssignedIdentities(ctx context.Context, specIdentities []infrav1.UserAssignedIdentity, vmIdentities []infrav1.UserAssignedIdentity) error {
+	expectedMap := make(map[string]struct{})
+	actualMap := make(map[string]struct{})
+
+	// Create a map of the expected identities. The ProviderID is converted to match the format of the VM identity.
+	for _, expectedIdentity := range specIdentities {
+		expectedClientID, err := s.identitiesGetter.GetClientID(ctx, expectedIdentity.ProviderID)
+		if err != nil {
+			return errors.Wrap(err, "failed to get client ID")
```

Comment:
It's probably worth adding a quick test case exercising this code path for that one extra line of sweet, sweet code coverage (and the actual benefits of verifying and documenting the behavior in this scenario).